server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sJobTemplate.java

```diff
@@ -137,4 +146,18 @@ private String getJobDefaultTemplate() throws IOException {
                 .getResourceAsStream(file);
         return new String(is.readAllBytes(), StandardCharsets.UTF_8);
     }
+
+    private void addDeviceInfoLabel(V1ObjectMeta meta, Map<String, ContainerOverwriteSpec> specs) {
+        if (meta == null) {
+            return;
+        }
+
+        var useGpu = specs.values().stream().anyMatch(spec ->
+                spec.resourceOverwriteSpec != null && spec.resourceOverwriteSpec.isRequireGpu());
+
+        if (meta.getLabels() == null) {
+            meta.labels(new HashMap<>());
+        }
+        meta.getLabels().put(deviceLabelName, useGpu ? "gpu" : "cpu");
```

Comment:
how about 
```Java
for (String device : resourceOverwriteSpec.getResourceSelector.getRequests().keSet()){
    meta.getLabels().put(deviceLabelName+device, true);
}
```
No change in `ResourceOverwriteSpec.java`

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
> Labels are key/value pairs. Valid label keys have two segments: an optional prefix and name, separated by a slash (/). 

It will generate a label with the key `device.starwhale.ai/nvidia.com/gpu` which is invalid.

Test with the command below
```k -n jialei label pod jialei-starwhale-controller-6cbbc69b4d-mjfdr device.starwhale.ai/nvidia.com/gpu=true```

> The Pod "jialei-starwhale-controller-6cbbc69b4d-mjfdr" is invalid: metadata.labels: Invalid value: "device.starwhale.ai/nvidia.com/gpu": a qualified name must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')

Reply:
Done